src/Forum/Content/User.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Forum\Content;
+
+use Flarum\Api\Client;
+use Flarum\Api\Controller\ShowUserController;
+use Flarum\Frontend\Document;
+use Flarum\Http\Exception\RouteNotFoundException;
+use Flarum\Http\UrlGenerator;
+use Flarum\User\User as FlarumUser;
+use Illuminate\Support\Arr;
+use Psr\Http\Message\ServerRequestInterface as Request;
+
+class User
+{
+    /**
+     * @var Client
+     */
+    protected $api;
+
+    /**
+     * @var UrlGenerator
+     */
+    protected $url;
+
+    /**
+     * @param Client $api
+     * @param UrlGenerator $url
+     */
+    public function __construct(Client $api, UrlGenerator $url)
+    {
+        $this->api = $api;
+        $this->url = $url;
+    }
+
+    public function __invoke(Document $document, Request $request)
+    {
+        $queryParams = $request->getQueryParams();
+        $actor = $request->getAttribute('actor');
+        $userId = Arr::get($queryParams, 'username');
+
+        $params = [
+            'id' => $userId,
+        ];
+
+        $apiDocument = $this->getApiDocument($actor, $params);
+
+        $document->title = $apiDocument->data->attributes->title;
+        $document->canonicalUrl = $this->url->to('forum')->route('user', ['username' => $userId]);
```

Comment:
Is that really the URL we want to be canonical?. At least in the frontend, we're always building the URL with the username.

That said, this has lead to problems (#1356), but we probably want to fix them by using a unique *slug* instead: https://github.com/flarum/core/issues/1734#issuecomment-570601659. So, using the username would be the closest to that desired reality for now, IMO.

Reply:
So instead of using the user-provider ID/username (as either works) we want to stick to the username? Ok, will do.

Reply:
Done. Changed it to `username`.